rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/rest/TbHttpClient.java

```diff
@@ -266,6 +268,16 @@ private TbMsg processException(TbContext ctx, TbMsg origMsg, Throwable e) {
             metaData.putValue(STATUS, httpClientErrorException.getStatusText());
             metaData.putValue(STATUS_CODE, httpClientErrorException.getRawStatusCode() + "");
             metaData.putValue(ERROR_BODY, httpClientErrorException.getResponseBodyAsString());
+        } else if (e instanceof HttpServerErrorException) {
```

Comment:
Use basic OOP principles to improve this code.  I believe it is sufficient to replace 
```
        if (e instanceof HttpClientErrorException) {
            HttpClientErrorException httpClientErrorException = (HttpClientErrorException) e;
```
with 
```
        if (e instanceof RestClientResponseException) {
            RestClientResponseException restClientResponseException = (RestClientResponseException) e;
```

Reply:
fix